golang/cosmos/types/conv/string.go

```diff
@@ -0,0 +1,27 @@
+package conv
+
+import (
+	"encoding/json"
+)
+
+// UnmarshalJSONString parses the JSON-encoded string data into v
+// See "encoding/json".Unmarshal
+func UnmarshalJSONString(data string, v any) error {
```

Comment:
Nit: "JSONText" names would be more accurate than "JSONString", cf. [RFC 8259](https://www.rfc-editor.org/rfc/rfc8259#section-2). I realize that "String" here can also be understood as the Go data type, but that's already inherent to their signatures. You could also just drop "String", leaving `UnmarshalJSON`, `ValidJSON`, and `MarshalToJSON`.